### PR TITLE
Add basic auth for server static

### PR DESCRIPTION
**Pull request by tuanpmt**

### CODE REVIEW

File: src/WebHandlerImpl.h [hunk 1]
```diff
@@ -49,6 +51,7 @@ class AsyncStaticWebHandler: public AsyncWebHandler {
     AsyncStaticWebHandler& setCacheControl(const char* cache_control);
     AsyncStaticWebHandler& setLastModified(const char* last_modified);
     AsyncStaticWebHandler& setLastModified(struct tm* last_modified);
+    AsyncStaticWebHandler& setBasicAuth(const char *username, const char *password);
```
Comment (me-no-dev) [inline src/WebHandlerImpl.h]: let's rename this to ```setAuthentication```  :)

**Comment by tuanpmt** [comment]
Maybe someone need websocket authentication too
```cpp
server
 .serveStatic("/", SPIFFS, "/")
 .setDefaultFile("index.html")
 .setAuthentication("user", "pass");
AsyncWebSocket ws("/ws");
ws.onEvent(onWsEvent);
ws.setAuthentication("user", "pass");
server.addHandler(&ws);
server.begin();
```

**Comment by me-no-dev** [comment]
Maybe event source too? And @tuanpmt please update README.md as well :)

**Comment by me-no-dev** [comment]
Great :) Thanks

**Comment by omersiar** [comment]
How this is work? I can still make unauthorized connection to WebSocket by going to 

http://www.websocket.org/echo.html

```bash
ws://ip_address/ws 
```
and click connect.

even if i set `ws.setAuthentication("admin", "adminpass");`

**Comment by omersiar** [comment]
Ok this was removed by further commit 

https://github.com/me-no-dev/ESPAsyncWebServer/commit/1b35f15ec24cb971c2bccbaa4810f08a10a5cd6e#diff-cda612710bbf9a4d4a36636d8ebb8264

**Comment by omersiar** [comment]
@me-no-dev Removing auth support were mandatory for improvement or it was by accident?

**Comment by omersiar** [comment]
I now understand it must be by accident because the other files other than the websocket are including this patch.

**Pull request merged**